File: youmai-sellergoods-interface/src/main/java/com/youmai/sellergoods/service/BrandService.java
package com.youmai.sellergoods.service;

import com.youmai.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {

    /**
     * 查询所有品牌
     *
     * @return
     */
    public List<TbBrand> findAll();



    public PageResult findPage(Integer pageNum, Integer pageSize);

    public PageResult findPage(TbBrand tbBrand,Integer pageNum, Integer pageSize);

    public void add(TbBrand tbBrand);

    public TbBrand findOne(Long id);

    public void update(TbBrand tbBrand);

    void delete(Long[] ids);

    List<Map> selectOptionList();
}

File: youmai-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务
app.service("brandService", function ($http) {
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    };

    this.findPage = function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size)
    };

    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    };

    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    };
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    };

    this.deleteBrands = function (selectIds) {
        return $http.get("../brand/delete.do?ids=" + selectIds);
    };

    this.search = function (page, size, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size, searchEntity);
    }
});
